substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/SubstrateOptions.java

```diff
@@ -202,6 +202,10 @@ protected void onValueUpdate(EconomicMap<OptionKey<?>, Object> values, String ol
     public static OptionEnabledHandler<Boolean> imageLayerEnabledHandler;
     public static OptionEnabledHandler<Boolean> imageLayerCreateEnabledHandler;
 
+    @APIOption(name = "-javaagent", valueSeparator = ':')//
+    @Option(help = "Enable the specified java agent in native image. Usage: -javaagent:<jarpath>[=<options>]. It's the same as using javaagent in JVM", type = User, stability = OptionStability.EXPERIMENTAL)//
+    public static final HostedOptionKey<AccumulatingLocatableMultiOptionValue.Strings> JavaAgent = new HostedOptionKey<>(AccumulatingLocatableMultiOptionValue.Strings.build());
```

Comment:
This partially conflicts with the basic JVMTI runtime support that was merged a couple weeks ago (see https://github.com/oracle/graal/pull/9558). I will need to think about that a bit and I will try to come up with an approach to unify the JVMTI build- and runtime support.

Reply:
Where does the conflict come from? Feels to me that with `javaagent` transformations happen mostly at build time (except the `premain`), while JVMTI provides hooks for run-time events.

Reply:
I don't think there is a conflict, as you said they are working on different phases. Class transformation has nothing to do with JVMTI events.